File: mqtt/src/main/java/org/sheedon/mqtt/ReadyTask.java
package org.sheedon.mqtt;

/**
 * @Description: 请求任务类
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2020/2/13 12:51
 */
public class ReadyTask {
    private String id;// 任务UUID
    private String backName; // 反馈名称
    private Callback callback;// 反馈监听器
    private DelayEvent event;// 超时处理事件


    public static ReadyTask build(String id, String backName,
                                  Callback callback, DelayEvent event) {
        ReadyTask task = new ReadyTask();
        task.id = id;
        task.backName = backName;
        task.callback = callback;
        task.event = event;
        return task;
    }

    public String getId() {
        return id;
    }

    public String getBackName() {
        return backName;
    }

    public Callback getCallback() {
        return callback;
    }

    public DelayEvent getEvent() {
        return event;
    }
}

File: mqtt/src/main/kotlin/org/sheedon/mqtt/internal/concurrent/ReadyTask.kt
package org.sheedon.mqtt.internal.concurrent

import org.sheedon.mqtt.IBack
import org.sheedon.mqtt.internal.connection.Listen

/**
 * 准备好的任务，包含的内容请求数据和反馈Callback
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2022/1/9 2:58 下午
 */
data class ReadyTask(
    var listen: Listen, // Call/Subscribe 的引用
    var id: Long, // 请求记录ID
    val type: CallbackEnum = CallbackEnum.SINGLE,//单次请求
    var back: IBack? = null // 反馈Callback
)